src/Core/Silk.NET.Core/Native/Structs/WinString.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Silk.NET.Core.Native;
+
+public unsafe struct WinString : IDisposable
+{
+    public struct Header
+    {
+        public uint Flags;
+        public uint Length;
+        public uint Reserved0;
+        public uint Reserved1;
+        public ushort* Data;
+        public int RefCount;
+        public ushort BufferStart;
+    }
+    public Header* HString;
+
+    public WinString(string str) : this((nuint)str.Length)
+        => str.AsSpan().CopyTo(new Span<char>(HString->Data, str.Length));
+
+    public WinString(WinString other) : this(other.HString->Length)
+        => new Span<ushort>(other.HString->Data, (int) other.HString->Length)
+            .CopyTo(new Span<ushort>(HString->Data, (int) other.HString->Length));
+
+    public WinString(nuint nChars)
+    {
+        [DllImport("kernel32.dll")]
+        static extern void* HeapAlloc(void* heap, uint dwFlags, nuint dwBytes);
+
+        HString = (Header*)HeapAlloc(GetProcessHeap(), 0, (nuint)sizeof(Header) + 2 * nChars);
+        HString->Flags = 0;
+        HString->Length = (uint)nChars;
+        HString->Reserved0 = 0;
+        HString->Reserved1 = 0;
+        HString->Data = &HString->BufferStart;
+        HString->RefCount = 1;
```

Comment:
Is there a reason we are using `HeapAlloc` specifically here? If we are required to, id prefer if we cache GetProcessHeap somewhere, to remove the need for the call to it *every* WinString creation

Reply:
_Technically_ you can replace `malloc`, though please don't. This is for consistency with the windows-rs implementation of HSTRING.